### PR TITLE
Added Kaggle melanoma example

### PR DESCRIPTION
**Pull request by mjang2000**

### CODE REVIEW

File: examples/vision/melanoma_classification_with_tfrecords.py [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+"""
+## 2.2 Data augmentation
+
+Image augmentation introduces new images into our data, and generally prevents the model
+from overfitting and leads to higher accuracy on testing datasets. For this particular
+dataset, we will randomly flip the image horizontally.
+"""
+
+
+def augmentation_pipeline(image, label):
```
Comment (fchollet) [inline examples/vision/melanoma_classification_with_tfrecords.py:173]: You would typically apply augmentation via preprocessing layers (applied after batching). See: https://keras.io/guides/preprocessing_layers/#image-data-augmentation-ondevice

Or https://keras.io/examples/vision/image_classification_from_scratch/#using-image-data-augmentation

File: examples/vision/melanoma_classification_with_tfrecords.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+"""
+"""
+# 1. Introduction + Set Up
+
+Transfer learning in computer vision is the process of using a pre-trained model on a new
+dataset of images. This is extremely useful when the dataset is not that big. ML models
+thrive when there is a lot of data available to train on, but it many scenarios,
+obtaining large datasets is quite difficult. The pre-trained model is trained on a big
+dataset, and the weights are preserved. The trained layers of the model is used as the
+basis for our new model. We will be using the VGG16 model to predict whether or not a
```
Comment (fchollet) [inline examples/vision/melanoma_classification_with_tfrecords.py:16]: VGG16 is fairly outdated (2014). Prefer using EfficientNet, ResNet50, or Xception for transfer learning

File: examples/vision/melanoma_classification_with_tfrecords.py [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+Title: Melanoma Classification with TFRecords
+Author: Amy MiHyun Jang
+Date created: 2020/07/29
+Last modified: 2020/07/29
+Description: Classifying skin lesion images from TFRecords for melanoma diagnosis
+"""
+"""
+# 1. Introduction + Set Up
+
+Transfer learning in computer vision is the process of using a pre-trained model on a new
```
Comment (fchollet) [inline examples/vision/melanoma_classification_with_tfrecords.py:11]: We have [a guide on transfer learning](https://keras.io/guides/transfer_learning/), as well as [a code example based on EfficientNet](https://keras.io/examples/vision/image_classification_efficientnet_fine_tuning/). It appears the main contribution of this example is to show how to read TFRecords with tf.data. Perhaps we could have a very streamlined version of this example under the section "quick Keras recipes", focused entirely on the TFRecord part of the workflow? "How to train a Keras model on TFRecord files"

**Comment by mjang2000** [comment]
This sounds like a great idea! I cut down on the other sections of the NB (no data augmentation, less explanations) for the latter half of the notebook to emphasize the TFRecord section.

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+"""
+## Introduction + Set Up
+
+TFRecords store a sequence of binary records, read linearly. They are useful format for
+storing data because they can be read efficiently. We'll explore how we can easily load
+in TFRecords for our melanoma classifier.
+"""
+
+"""shell
+! pip install gcsfs -q
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:17]: Do not include `!` in shell blocks, it will be automatically added

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+Title: How to train a Keras model on TFRecord files
+Author: Amy MiHyun Jang
+Date created: 2020/07/29
+Last modified: 2020/08/06
+Description: Loading TFRecords for computer vision models
+"""
+"""
+## Introduction + Set Up
+
+TFRecords store a sequence of binary records, read linearly. They are useful format for
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:11]: Please add a link to a page where the user can find out more about TFRecords

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+"""
+
+import re
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from functools import partial
+import matplotlib.pyplot as plt
+
+try:
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:27]: I recommend remove all TPU-related code, as this is outside the scope of TFRecord loading

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+GCS_PATH = "gs://kds-f809fea39df606e89617cd8d8e4cacc083fb6e176982f0e5d69215a8"
+BATCH_SIZE = 16 * strategy.num_replicas_in_sync
+IMAGE_SIZE = [1024, 1024]
+
+"""
+## Load the data
+
+Now we will load in our data. For this notebook, we will be importing the TFRecord files.
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:52]: "Now we will load in our data." -> already conveyed by the section title

" For this notebook, we will be importing the TFRecord files." -> already conveyed by the page title

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+It is good practice to divide the training set data into two. The smaller dataset will be
+the validation set. Having a validation set is useful to slow down overfitting.
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:54]: "Having a validation set is useful to slow down overfitting." -> having a validation set is critical whatever you do, so you can remove this sentence

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+
+We also reshape our data so that all of the images will be the same shape. Although the
+TFRecord files have already been reshaped for us, it is best practice to reshape the
+input so that we know exactly what's going in to our model.
+"""
+
+
+def decode_image(image):
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.cast(image, tf.float32) / 255.0
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:84]: Since you're using Xception, you should do this via `applications.xception.preprocess_input`. Indeed, different applications use different value normalization scheme, so it's better to always import the one that corresponds to the model you're using.

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+    image = tf.reshape(image, [*IMAGE_SIZE, 3])
+    return image
+
+
+"""
+As we load in our data, we need both our ```X``` and our ```Y```. The X is our image; the
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:90]: Only one backtick needed, not 3

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+        total_img, malignant, 100 * malignant / total_img
+    )
+)
+
+"""
+### Correcting for data inbalance
+"""
+
+"""
+#### Set initial bias
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:203]: This will be immediately learned by the model and thus shouldn't be necessary.

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+weight_for_0 = (1 / benign) * (total_img) / 2.0
+weight_for_1 = (1 / malignant) * (total_img) / 2.0
+
+class_weight = {0: weight_for_0, 1: weight_for_1}
+
+print("Weight for class 0: {:.2f}".format(weight_for_0))
+print("Weight for class 1: {:.2f}".format(weight_for_1))
+
+"""
+## Building our model
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:230]: Remove?

File: examples/keras_recipes/melanoma_classification_with_tfrecords.py [hunk 13]
```diff
@@ -0,0 +1,336 @@
...
+    )
+
+    base_model.trainable = False
+
+    model = tf.keras.Sequential(
+        [
+            base_model,
+            tf.keras.layers.GlobalAveragePooling2D(),
+            tf.keras.layers.Dense(
+                8, activation="relu", kernel_regularizer=tf.keras.regularizers.l2(0.001)
```
Comment (fchollet) [inline examples/keras_recipes/melanoma_classification_with_tfrecords.py:283]: L2 regularization should not be necessary here

**Comment by mjang2000** [comment]
The NB convert did not work with the original file because the kernel died from exceeding the RAM limit. I included the TPU-related code to fix this problem, but please let me know if there is another way
